src/pkcs15init/pkcs15-epass2003.c

```diff
@@ -659,7 +659,27 @@ static int epass2003_pkcs15_generate_key(struct sc_profile *profile,
 		rsa->exponent.len = 3;
 
 		pubkey->algorithm = SC_ALGORITHM_RSA;
-	} else
+	}
+	else if(pubkey && obj->type == SC_PKCS15_TYPE_PRKEY_EC)
+	{
+		struct sc_ec_parameters *ecparams = (struct sc_ec_parameters *)key_info->params.data;
+		pubkey->algorithm = SC_ALGORITHM_EC;
+		pubkey->u.ec.ecpointQ.value = malloc(65);
+		memcpy(pubkey->u.ec.ecpointQ.value, gendat.modulus, 65);
```

Comment:
error check on `malloc` is missing.

is `gendat.modulus` garanteed to hold 65 bytes?

Reply:
I wonder where this magic number `65` is coming from. But since this is for a specific card, perhaps it's of no significance...?

Reply:
https://tools.ietf.org/id/draft-jivsov-ecc-compact-05.html#overview spells it out.
for a 256 curve, one byte compact flag ( usually 0x04) and x and y (256 bits each) or 32 bytes each for a total of 65 bytes.

They will have to change this if they support any other sizes.